opal/mca/common/ucx/common_ucx.c

```diff
@@ -74,3 +74,70 @@ OPAL_DECLSPEC void opal_common_ucx_mca_pmix_fence(ucp_worker_h worker)
     }
 }
 
+
+static void opal_common_ucx_wait_all_requests(void **reqs, int count,
+                                              ucp_worker_h worker)
+{
+    int i;
+
+    MCA_COMMON_UCX_VERBOSE(2, "waiting for %d disconnect requests", count);
+    for (i = 0; i < count; ++i) {
+        opal_common_ucx_wait_request(reqs[i], worker, "ucp_disconnect_nb");
+        reqs[i] = NULL;
+    }
+}
+
+OPAL_DECLSPEC int opal_common_ucx_del_procs(opal_common_ucx_del_proc_t *procs, size_t count,
+                                            size_t max_disconnect, ucp_worker_h worker)
+{
+    int num_reqs;
+    size_t max_reqs;
+    void *dreq, **dreqs;
+    size_t i;
+
+    MCA_COMMON_UCX_ASSERT(procs || !count);
+
+    max_reqs = max_disconnect ? max_disconnect : 1;
```

Comment:
why?

Reply:
because there is no check if max_disconnect is NULL, replaced by assert